app/parse.py

```diff
@@ -12,7 +33,94 @@ class Course:
     name: str
     short_description: str
     course_type: CourseType
+    modules: int
+    topics: int
+    duration: str
+
+
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def get_course_additional_details(course_detail_url: str) -> dict:
+    course_detail_page = requests.get(
+        urljoin(BASE_URL, course_detail_url)
+    ).content
+    course_detail_soup = BeautifulSoup(course_detail_page, "html.parser")
+
+    course_additional_details = dict(
+        modules=int(course_detail_soup.select_one(
+            ".CourseModulesHeading_modulesNumber__GNdFP"
+        ).text.split()[0]),
+        topics=int(course_detail_soup.select_one(
+            ".CourseModulesHeading_topicsNumber__PXMnR"
+        ).text.split()[0]),
+        duration=course_detail_soup.select_one(
+            ".CourseModulesHeading_courseDuration__f_c3H"
+        ).text,
+    )
+
+    return course_additional_details
+
+
+def get_single_course_full_details(course: Tag) -> Course:
+
+    course_detail_url = course.select_one(".mb-16")["href"]
+    course_type_in_url = course.select_one("a.mb-16")["href"].split("-")[-1]
+
+    course_name = course.select_one(".typography_landingH3__vTjok").text
+    logging.info(f"Parsing course: {course_name}")
+
+    course_base_details = dict(
+        name=course_name,
+        short_description=course.select_one(
+            "p.CourseCard_courseDescription__Unsqj"
+        ).text,
+        course_type=(
+            CourseType.PART_TIME
+            if course_type_in_url == "parttime"
+            else CourseType.FULL_TIME
+        )
+    )
+    course_additional_details = get_course_additional_details(
+        course_detail_url
+    )
+
+    return Course(
+        **course_base_details,
+        **course_additional_details
+    )
```

Comment:
Move this as class method to Course class

Reply:
> Move this as class method to Course class

Noted and will be done. 
Is better to make function get_course_additional_details as static method of Course in this case?